distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFtp.java

```diff
@@ -264,4 +276,145 @@ private void showServerReply(FTPClient ftpClient) {
+        safeSetClientAuthConfiguration("want");
+        assertTrue(testSslHandshake(true));
+    }
+
+    /**
+     * Client with keystore should be able to connect and successfully complete SSL handshake
+     * with FTP server when clientAuth = "need"
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testFtpsSslHandshakeNeedClientAuthWithKeystore() throws Exception {
+        safeSetClientAuthConfiguration("need");
+        assertTrue(testSslHandshake(true));
+    }
+
+    /**
+     * Client without keystore should be able to connect and successfully complete SSL handshake
+     * with FTP server when clientAuth = "want"
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testFtpsSslHandshakeWantClientAuthWithoutKeystore() throws Exception {
+        safeSetClientAuthConfiguration("want");
+        assertTrue(testSslHandshake(false));
+    }
+
+    /**
+     * Client without keystore should be able to connect but fail to complete SSL handshake
+     * with FTP server when clientAuth = "need"
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testFtpsSslHandshakeNeedClientAuthWithoutKeystore() throws Exception {
+        safeSetClientAuthConfiguration("need");
+        assertFalse(testSslHandshake(false));
+    }
+
+    private boolean testSslHandshake(Boolean setKeystore) throws Exception {
+        FTPSClient ftps = new FTPSClient();
+
+        if (setKeystore) {
+            KeyManager keyManager =
+                    KeyManagerUtils.createClientKeyManager(new File(System.getProperty(
+                            "javax.net.ssl.keyStore")),
+                            System.getProperty("javax.net.ssl.keyStorePassword"));
+            ftps.setKeyManager(keyManager);
+        }
+
+        try {
+            ftps.connect(FTP_SERVER, FTP_DEFAULT_PORT);
+            showServerReply(ftps);
+            int connectionReply = ftps.getReplyCode();
+            if (!FTPReply.isPositiveCompletion(connectionReply)) {
+                fail("FTP server refused connection: " + connectionReply);
+            }
+
+            boolean success = ftps.login(USERNAME, PASSWORD);
+            showServerReply(ftps);
+            if (!success) {
+                fail("Could not log in to the FTP server.");
+            }
+
+            ftps.enterLocalPassiveMode();
+            ftps.setControlKeepAliveTimeout(300);
+            ftps.setFileType(FTP.BINARY_FILE_TYPE);
+
+            disconnectClient(ftps);
+            return true;
+        } catch (SSLException e) {
+            disconnectClient(ftps);
+            return false;
+        }
+    }
+
+    /**
+     * Sets the clientAuth configuration in catalog-ftp feature.
+     *
+     * @throws Exception
+     */
+    private void safeSetClientAuthConfiguration(String clientAuth) throws Exception {
+        switch (clientAuth) {
+        case "want":
+            setClientAuthConfiguration("need");
+            setClientAuthConfiguration("want");
```

Comment:
Shouldn't these options just do one or the other? I guess I'm not seeing the value on setting it to the opposite one before setting it to the desired option.